docs/languages/csharp/expression-trees.md

```diff
@@ -0,0 +1,132 @@
+﻿# Expression Trees
+Expression Trees are one of the core concepts in computer science language deign.
+In most languages you may have used, expression trees are used by the language
+compiler software as part of the compilation process.
+
+However, many popular languages hide the details of expression trees from the
+developers that use the language.
+
+C# is different.
+
+C# developers can create and interact with Expression Trees.
+
+This article covers the fundamentals of expression trees, how they are created
+and executed, and how they can be an important tool for understanding
+code.
+## Defining Expression Trees
+Expression Trees are a data structure that defines your code. Let's look at a simple example.
+Here's a line of code:
+```cs
+var sum = 1 + 2;
+```
+If you were to analyze this as an expression tree, it contains several nodes.
+The outermost node is a variable declaration statement with assignment (`var sum = 1 + 2;`)
+That outermost node contains several child nodes: a variable declaration, an assignment node, and an
+expression representing the right hand side of the equals sign.
+
+Let's drill down a bit more into the expressions that make up the right side of the equals sign.
+The expression is `1 + 2`. That's a binary expression. More specifically, it's a binary addition
+expression. A binary addition expression has two children, representing the left and right nodes
+of the addition expression. Here, both nodes are constant expressions: The left operand is the
+value `1`, and the right operand is the value `2`.
+
+Visually, the entire statement is a tree: You could start at the root node, and travel to
+each node in the tree to see the code that makes up the statement:
+
+- Variable declaration statement with assignment
+    * Variable Declaration (`var sum = 1 + 2`)
+        - Implicit variable type declaration (`var sum`)
+            * Implicit var keyword (`var`)
+            * Variable name declaration (`sum`)
+    * assignment expression (`=`)
+    * Binary Addition Expression (`1 + 2`)
+        - left operand (`1`)
+        - addition operator (`+`)
+        - right operand (`2`)
+
+This may look complicated, but it is very powerful. Following the same process, you can decompose
+much more complicated expressions. Consider this expression:
+```cs
+var finalAnswer = this.SecretSauceFuncion(
+    currentState.createInterimResult(), currentState.createSecondValue(1, 2),
+    decisionServer.considerFinalOptions("hello")) +
+    MoreSecretSauce('A', DateTime.Now, true);
+```
+
+The expression above is also a variable declaration with an assignment.
+In this instance, the right hand side of the assignment is a much more complicated tree.
+I'm not going to decompose this expression, but consider what the different nodes might
```

Comment:
This feels a little weird w.r.t the voice of the article.  Perhaps just the latter part with a minor adjustment "Consider what the different nodes might look like" is good enough for this sentence?  I tend to react slightly negatively to documentation when the author says they're not going to do something, so if I read this I might be inclined to just pass over the content.